rest/multipart.go

```diff
@@ -37,10 +37,23 @@ const kMaxInlineAttachmentSize = 200
 // JSON bodies smaller than this won't be GZip-encoded.
 const kMinCompressedJSONSize = 300
 
-// ReadJSONFromMIME parses a JSON MIME body, unmarshalling it into "into".
+// ReadJSONFromMIMEWithErrorMessage parses a JSON MIME body, unmarshalling it into "into".
 // Closes the input io.ReadCloser once done.
+func ReadJSONFromMIMEWithErrorMessage(headers http.Header, input io.ReadCloser, into interface{}) error {
```

Comment:
Would prefer to keep the error messages built in the existing `ReadJSONFromMIME` function, and have a "raw" version, given the rarer use-case for the latter.